### ISSUE TITLE
numpy 1.16

### ISSUE THREAD

**Issue by stevenwong** [fo40225/Anaconda-Windows-AMD#6]
Hi, thanks for uploading this. I'm trying to install tensorflow 1.14 which supposedly requires numpy 1.16. How did you patch numpy?

**Comment by nicolaerosia** [comment]
@fo40225 can you share the patching procedure? I would like to do this for Linux

**Comment by ValZapod** [comment]
@fo40225 Have you seen this MKL_DEBUG_CPU_TYPE=5 https://www.reddit.com/r/matlab/comments/dxn38s/howto_force_matlab_to_use_a_fast_codepath_on_amd/****

**Comment by fo40225** [comment]
> @fo40225 can you share the patching procedure? I would like to do this for Linux

https://docs.scipy.org/doc/numpy/user/building.html

https://software.intel.com/en-us/articles/numpyscipy-with-intel-mkl

https://github.com/fo40225/Anaconda-Windows-AMD/blob/master/site.cfg

https://www.agner.org/optimize/#asmlib

**Comment by fo40225** [comment]
> @fo40225 Have you seen this MKL_DEBUG_CPU_TYPE=5 https://www.reddit.com/r/matlab/comments/dxn38s/howto_force_matlab_to_use_a_fast_codepath_on_amd/

Looks good, haven't seen before.

I will try it.

**Comment by ValZapod** [comment]
@fo40225 Yeah, cool. Also note some things. https://www.mathworks.com/matlabcentral/answers/330889-does-matlab-perform-well-on-amd-ryzen
The problem was in 2017 Ryzens broking with FMA3 commands. (Now it is good.) Also see https://gmplib.org/

> Ryzen CPU caution: The AMD Ryzen 5 and 7 CPUs sold up until at least November 2017 have serious bugs which make any software run very unreliably. One experiences spurious segfaults and system hangs from building as well as running GMP (or any other CPU intensive stuff).

Also there is description of it https://github.com/SherbazHashmi/HackathonServer/blob/14532503b9a009474a27444177e33fc0ab9f6b76/arcpyenv/arcgispro-py3-clone/Lib/site-packages/despatch/despatch.py#L60 but note there is also 6 and 7 (AVX512 is not on AMD, obviously).
https://github.com/chameleonTK/cloud-recommendation-kubernetes/blob/8275cf5646ec71d2bc8fa6e68bb24affc0d4449a/PerfKitBenchmarker/perfkitbenchmarker/linux_benchmarks/hpcc_benchmark.py#L94

Also please give another wikipedia edit here with results https://en.wikipedia.org/wiki/Math_Kernel_Library#Performance
(I already edited in basic info about it just now.)

**Comment by fo40225** [comment]
On zen or zen+ architecture (ryzen 1000 or 2000 series), since the floating-point arithmetic unit is 128 bits wide, the 256 bits AVX2 and FMA3 instruction sets will not have double the throughput.

**Comment by ValZapod** [comment]
@fo40225 Are you sure)) I thought AVX2 at least has some standard behaviour, without any connection to FPU. So, anyway, is there any speed gain?
